src/server.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+import * as ReactDOMServer from "react-dom/server";
+import { load }  from "cheerio";
+import { PortalManager, ChildrenSelectorTuple } from "./PortalManager";
+
+export class ServerPortal {
+  portals: ChildrenSelectorTuple[];
+
+  constructor() {
+    this.portals = [];
+    this.appendUniversalPortals = this.appendUniversalPortals.bind(this);
+  }
+
+  collectPortals(children: React.ReactNode) {
+    return <PortalManager portals={this.portals}>{children}</PortalManager>;
+  }
+
+  appendUniversalPortals(html: string) {
```

Comment:
You can 

```
appendUniversalPortals = (html:string) {
```

And avoid 

```
this.appendUniversalPortals = this.appendUniversalPortals.bind(this);
```

in the constructor

Reply:
Great suggestion, thanks!